src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTestService.swift

```diff
@@ -698,7 +699,7 @@ class CoronaTestService {
 
 					}
 
-					if coronaTest.finalTestResultReceivedDate == nil {
+					if self.coronaTest(ofType: coronaTestType)?.finalTestResultReceivedDate == nil {
```

Comment:
@ArturFriesen This was the reason the test was failing. It was referencing a stale value. With this change it now works as expected.

Reply:
Ok nice!

Reply:
Referring to @KaiTeuber comment, this shows, that there is still the problem with stale data.